File: DeadZone-Beta/webserver/scripts/masterFile.js
/*
    Author: Levi Smith
    Version: Alpha
    Date: 07/29/2025
*/

export class MasterFile {
    constructor() {
        this.file = {};
    }

    loadFile(file) {
        this.file = file;
    }

    export() {
        console.log("exporting master file");
        return JSON.stringify(this.file)
    }

    updateTower(tower) {
        this.file["tower" + tower.id]["color"] = tower.color;
        this.file["tower" + tower.id]["transition"] = tower.transition
        this.file["tower" + tower.id]["effect"] = tower.effect;
    }

    updateTowerColor(id, color) {
        this.file["tower" + id]["color"] = color;
    }

    updateTowerTransition(id, transition) {
        this.file["tower" + id]["transition"] = transition;
    }

    updateTowerEffect(id, effect) {
        this.file["tower" + id]["effect"] = effect;
    }

    getTowerColor(id) {
        return this.file["tower" + id]["color"]
    }

    getTowerTransition(id) {
        return this.file["tower" + id]["transition"]
    }

    getTowerEffect(id) {
        return this.file["tower" + id]["effect"]
    }

    getTowerStatus(id) {
        return this.file["tower" + id]["status"]
    }

}

File: DeadZone-Alpha/websocketserver/masterFile.py
"""
	Author: Levi Smith
	Version: Alpha
	Date: 07/29/2025
"""

import json
from defaults import DEFAULT_COLOR, DEFAULT_STATUS, DEFAULT_TRANSITION, DEFAULT_EFFECT
from colors import COLORS

class MasterFile:

	def __init__(self, numTowers):
		self.numTowers = numTowers
		self.doc = {}
		self.doc["type"] = "masterFile"
		for i in range(numTowers):
			self.resetTower(i + 1)
	
	def export(self):
		return json.dumps(self.doc)

	def exportTower(self, id):
		towerDoc = {
			"color": COLORS[self.doc["tower" + str(id)]["color"]],
			"transition": self.doc["tower" + str(id)]["transition"],
			"effect": self.doc["tower" + str(id)]["effect"]
		}
		return json.dumps(towerDoc)
	
	def getDoc(self):
		return self.doc
	
	def overwrite(self, doc):
		self.doc = doc
	
	def updateDoc(self, file):
		for i in range(self.numTowers):
			id = i + 1
			self.configureTower(id, file["tower" + str(id)]["color"], file["tower" + str(id)]["transition"], file["tower" + str(id)]["effect"])
			
	def resetTower(self, id):
		self.doc["tower" + str(id)] = {"status": DEFAULT_STATUS, "color": DEFAULT_COLOR, "transition":DEFAULT_TRANSITION, "effect":DEFAULT_EFFECT}
	
	def configureTower(self, id, color, transition, effect):
		self.doc["tower" + str(id)]["color"] = color
		self.doc["tower" + str(id)]["transition"] = transition
		self.doc["tower" + str(id)]["effect"] = effect

	def updateTower(self, id, status, color, transition, effect):
		self.doc["tower" + str(id)] = {"status": status, "color": color, "transition":transition, "effect":effect}
	
	def updateTowerStatus(self, id, status):
		self.doc["tower" + str(id)]["status"] = status

	def updateTowerColor(self, id, color):
		self.doc["tower" + str(id)]["color"] = color
	
	def updateTowerTransition(self, id, transition):
		self.doc["tower" + str(id)]["transition"] = transition
	
	def updateTowerEffect(self, id, effect):
		self.doc["tower" + str(id)]["effect"] = effect